packages/material-ui/scripts/rollup.config.js

```diff
@@ -41,7 +41,21 @@ const commonjsOptions = {
 };
 
 function onwarn(warning) {
-  throw Error(warning.message);
+  let ignoreWarning = false;
+  if (
+    warning.code === 'UNUSED_EXTERNAL_IMPORT' &&
+    warning.source === 'react' &&
+    warning.names.filter(identifier => identifier !== 'useDebugValue').length === 0
+  ) {
+    // skip
+    // import * as React from 'react'
+    // if (__DEV__) React.useDebugValue()
+    ignoreWarning = true;
```

Comment:
@TrySound Is this expected from rollup? Feels like it should not warn here but remove the import. Or is this `terser` territory?

Reply:
Let's leave this as a console.warn as opposed to throwing as a reminder that this isn't fully dce'd (Dead Code Elimination). It's only 23Bytes anyway.